torch/csrc/api/include/torch/linalg.h

```diff
@@ -21,4 +37,20 @@ inline Tensor linalg_det(const Tensor& self) {
   return detail::det(self);
 }
 
+inline Tensor linalg_norm(const Tensor& self, optional<Scalar> opt_ord, optional<IntArrayRef> opt_dim, bool keepdim, optional<ScalarType> opt_dtype) {
```

Comment:
Hey @mruberry, I wonder, do you think this whole file could potentially be autogened? It seems like the changes required to add a new function are fairly mechanical.

Reply:
I agree with you and that's a cool suggestion. @glaringlee who works on our C++ API for him to consider and prioritize it.